stdlib/iter.mli

```diff
@@ -0,0 +1,42 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                 Simon Cruanes                                          *)
+(*                                                                        *)
+(*   Copyright 2001 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(* Module [Iter]: functional iterators *)
+
+(** @since NEXT_RELEASE *)
+
+type ('a, 's) step =
+  | Done
+  | Skip of 's
+  | Yield of 'a * 's
+
+type +_ t =
+  | Sequence : 's * ('s -> ('a,'s) step) -> 'a t
```

Comment:
I think it's useful to be able to write one's own combinators based on the type definition. At the very least, it should be within a `(**/**)` block so that the various stdlib extensions/replacements can interoperate with it.